mantidproject/mantid
# Loader for new D16 data

**Description of work.**
This PR makes it possible for Mantid to manage the files D16 will output once its detector is changed next cycle. The format does not fundamentally change, but scans are now grouped into a multidimensional file using the TOF dimension for the scanned parameter.

Reduction algorithms usually used by D16 are also updated to reflect this change.



**To test:**
-  Load a test file ( such as `UnitTest/ILL/D16/066321.nxs` which is mostly white noise except for one corner) and check it is well formed
- Open the instrument viewer and check the view looks fine (especially in 3D with the proportions correct)



Closes #34163  






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
